decidim-elections/app/commands/decidim/votings/census/admin/create_dataset.rb

```diff
@@ -55,7 +55,7 @@ def create_census_dataset!
               current_user,
               {
                 voting: form.current_participatory_space,
-                file: form.file,
+                file: form.file.filename.to_s,
```

Comment:
> Do you think it would also make sense to store the blob ID and/or key to the admin logs? I think the filename information is still relevant e.g. for showing to the users but just thinking from the admin logging perspective to be able to track changes back.

Well, this thing is a little weird, this is called "file" but it is not a file (maybe this started using carrierwave and, when porting it to active storage this was left for reference). For now, it is only used as the filename to give the admin0 a reference of the uploaded file. The actual attached file is calle `acces_code_file`.
I guess we could even remove this column and extract the name directly from the blob table. Let me know and I'll do it if you want.

Reply:
> Another thing I also noticed is that the line counting is happening with a shell command wc at the file_lines_count method. Do you have any idea why it's not done in Ruby? Performance reasons, etc.? 

My guess is performance/memory, didn't check this file. I also thing we should avoid external tools for portability. In anycase, if it is really necessary it should have a comment explaining the reasons. I'll see what can be done here

Reply:
> Well, this thing is a little weird, this is called "file" but it is not a file (maybe this started using carrierwave and, when porting it to active storage this was left for reference). For now, it is only used as the filename to give the admin0 a reference of the uploaded file. The actual attached file is calle `acces_code_file`.
> I guess we could even remove this column and extract the name directly from the blob table. Let me know and I'll do it if you want.

Now that you say it, I agree that it is weird. This definitely seems like something that was left over from CarrierWave. I actually initially thought it was only meant for the action log purposes, but not I realized it's actually stored against the dataset record as you mentioned.

So definitely, let's remove any left overs from CarrierWave!

> My guess is performance/memory, didn't check this file. I also thing we should avoid external tools for portability. In anycase, if it is really necessary it should have a comment explaining the reasons. I'll see what can be done here

Great! And just to be more specific, we don't need to handle that in this PR unless you want to. We can also consider this as a separate issue.

Reply:
> So definitely, let's remove any left overs from CarrierWave!

I think it is not possible to retrieve the original name of the file using the Blog object. As an alternative I've renamed the `file` column to `filename` (maybe we could make it `original_filename` if you prefer.

Unless you have other ideas...

Reply:
> I think it is not possible to retrieve the original name of the file using the Blob object. As an alternative I've renamed the `file` column to `filename` (maybe we could make it `original_filename` if you prefer.
> 
> Unless you have other ideas...

There may be something I'm not understanding here but try this in Rails console:
```irb
> ActiveStorage::Blob.last.filename
=> "census-empty.csv"
```

Reply:
Well, here's the thing, it's not that the blob doesn't give you a filename, it does. The problem here is that this file is not the original uploaded by the admin. Is a file generated from the one uploaded: You upload a CSV, it gets processed and then attached as the `access_codes_file` (which is not a CSV btw, it's a ZIP).

Here is what you see in the console:
```ruby
Decidim::Votings::Census::Dataset.last

#<Decidim::Votings::Census::Dataset:0x000055b9df188cf8 id: 5, filename: "cens_9_persones.csv", status: "freeze", data_count: 9, csv_row_raw_count: 9, csv_row_processed_count: 9, decidim_votings_voting_id: 7, created_at: Thu, 07 Jul 2022 12:31:48.883489000 UTC +00:00, updated_at: Thu, 07 Jul 2022 12:56:44.013514000 UTC +00:00>

Decidim::Votings::Census::Dataset.last.access_codes_file.filename

#<ActiveStorage::Filename:0x000055b9dbd8ca08 @filename="kWPhJuI-YuwO7cpocEMj7A.zip">
```

I still think that this is a bit odd, due the fact that the original file is not conserved anywhere (but maybe that's is the way to go to garantee privacy). In any case, could we treat this as a separate issue in case you consider that is necessary to change this logic?

Reply:
I trust your insight regarding this, so let's keep it as-is. You know and understand this module much better than I do.

So let's not change it for now, I'm sure you've thought that through.